packages/generator/templates/app/jest.config.js

```diff
@@ -1,37 +1,29 @@
+const { pathsToModuleNameMapper } = require("ts-jest/utils")
+const { compilerOptions } = require("./tsconfig")
+
 module.exports = {
+  // Test setup file
+  setupFilesAfterEnv: ["<rootDir>/test/setup.ts"],
+  // Add type checking to Typescript test files
+  preset: "ts-jest",
+  testEnvironment: "jest-environment-jsdom-fourteen",
   // Automatically clear mock calls and instances between every test
   clearMocks: true,
+  testPathIgnorePatterns: ["/node_modules/", "/.blitz/", "/.next/", "<rootDir>/db/migrations"],
+  transformIgnorePatterns: ["[/\\\\]node_modules[/\\\\].+\\.(ts|tsx)$"],
+  transform: {
+    "^.+\\.(ts|tsx)$": "babel-jest",
+  },
+  // This makes absolute imports work
+  moduleDirectories: ["node_modules", "."],
+  moduleNameMapper: {
+    // This ensures any path aliases in tsconfig also work in jest
+    ...pathsToModuleNameMapper(compilerOptions.paths || {}),
+    "\\.(css|less|sass|scss)$": "identity-obj-proxy",
+    "\\.(gif|ttf|eot|svg|png|jpg|jpeg)$": "<rootDir>/test/__mocks__/fileMock.js",
+  },
+  watchPlugins: ["jest-watch-typeahead/filename", "jest-watch-typeahead/testname"],
   // Coverage output
-  coverageDirectory: '.coverage',
-  // Test setup file
-  setupFilesAfterEnv: ['./jest.setup.js'],
-  roots: [
-    "<rootDir>/app"
-  ],
-  collectCoverageFrom: [
-    "app/**/*.{js,jsx,ts,tsx}",
-    "!app/**/*.d.ts"
-  ],
-  testMatch: [
-    "<rootDir>/app/**/__tests__/**/*.{js,jsx,ts,tsx}",
-    "<rootDir>/app/**/*.{spec,test}.{js,jsx,ts,tsx}"
-  ],
-  testEnvironment: "jest-environment-jsdom-fourteen",
-  modulePaths: [],
-  moduleFileExtensions: [
```

Comment:
**question**: Why did you remove the file extensions to watch? Is that being replace by any combination of the rules you have added or we will track any file extension?

Reply:
The default includes js, jsx, ts, tsx, and node. And the extra entries with `web` are not needed from what I can tell.

Reply:
Awesome!